ui/src/main/java/org/pentaho/di/ui/i18n/MessagesSourceCrawler.java

```diff
@@ -88,115 +149,77 @@
    */
   private List<SourceCrawlerXMLFolder> xmlFolders;
 
-  private Pattern packagePattern;
-  private Pattern importPattern;
-  private Pattern importMessagesPattern;
-  private Pattern stringPkgPattern;
-  private Pattern classPkgPattern;
+  private Pattern packagePattern = Pattern.compile( PACKAGE_PATTERN );
+  private Pattern importPattern = Pattern.compile( IMPORT_PATTERN );
+  private Pattern importMessagesPattern = Pattern.compile( IMPORT_MESSAGES_PATTERN );
+  private Pattern stringPkgPattern = Pattern.compile( STRING_PKG_PATTERN );
+  private Pattern classPkgPattern = Pattern.compile( CLASS_PKG_PATTERN );
```

Comment:
If the constants above are only used here, why declare private constants?

Reply:
As you see, I've just inlined the assignment but you're right: as Pattern instances are immutable and already the regex expression used is a static final, these can also be static final!